src/controller/p2p.rs

```diff
@@ -259,7 +309,17 @@ impl<EventStream: Stream<Item = WorkerEvent>> P2PController<EventStream> {
                         }
                     }
                 }
-                WorkerEvent::LogsRequest { .. } => todo!(),
+                WorkerEvent::LogsRequest { request, .. } => {
+                    match self.log_requests_tx.try_send(request) {
+                        Ok(_) => {}
+                        Err(mpsc::error::TrySendError::Full(_)) => {
+                            warn!("There is an ongoing logs request already");
```

Comment:
Unlike to query processing, it is kinda ok to simply drop the request here, but the query length could be greater than 1, because a couple of Log collectors can hit us at the same time.

Reply:
Actually, it was not ok because the transport lib kept the connection open in this case :(
I've changed it to hold the response sender, which triggers closing the connection on drop.